src/parser/flow_lexer.ml

```diff
@@ -1579,46 +1727,62 @@ let type_token env lexbuf =
    * Number literals
    *)
 
-  | Opt neg, binnumber, (letter | '2'..'9'), Star alphanumeric ->
+  | Opt neg, binbigint ->
+    let num = lexeme lexbuf in
+    Token (env, mk_bignum_singleton BINARY num)
+  | Opt neg, (binnumber | binbigint), (letter | '2'..'9'), Star alphanumeric ->
     (* Numbers cannot be immediately followed by words *)
     recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
     | Opt neg, binnumber ->
       let num = lexeme lexbuf in
       Token (env, mk_num_singleton BINARY num)
     | _ -> failwith "unreachable"
     )
-
   | Opt neg, binnumber ->
     let num = lexeme lexbuf in
     Token (env, mk_num_singleton BINARY num)
 
-  | Opt neg, octnumber, (letter | '8'..'9'), Star alphanumeric ->
+  | Opt neg, octbigint ->
+    let num = lexeme lexbuf in
+    Token (env, mk_bignum_singleton OCTAL num)
+  | Opt neg, (octnumber | octbigint), (letter | '8'..'9'), Star alphanumeric ->
     (* Numbers cannot be immediately followed by words *)
     recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
     | Opt neg, octnumber ->
       let num = lexeme lexbuf in
       Token (env, mk_num_singleton OCTAL num)
     | _ -> failwith "unreachable"
     )
-
   | Opt neg, octnumber ->
     let num = lexeme lexbuf in
     Token (env, mk_num_singleton OCTAL num)
 
-  | Opt neg, legacyoctnumber, (letter | '8'..'9'), Star alphanumeric ->
+  | Opt neg, legacyoctbigint ->
+    let num = lexeme lexbuf in
+    Token (env, mk_bignum_singleton LEGACY_OCTAL num)
+  | Opt neg, (legacyoctnumber | legacyoctbigint), (letter | '8'..'9'), Star alphanumeric ->
     (* Numbers cannot be immediately followed by words *)
     recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
     | Opt neg, legacyoctnumber ->
       let num = lexeme lexbuf in
       Token (env, mk_num_singleton LEGACY_OCTAL num)
     | _ -> failwith "unreachable"
     )
-
   | Opt neg, legacyoctnumber ->
     let num = lexeme lexbuf in
     Token (env, mk_num_singleton LEGACY_OCTAL num)
 
-  | Opt neg, hexnumber, non_hex_letter, Star alphanumeric ->
+  | Opt neg, hexbigint ->
+    let num = lexeme lexbuf in
+    let loc = loc_of_lexbuf env lexbuf in
+    let env = lex_error env loc Parse_error.InvalidBigInt in
```

Comment:
I can't figure out why this is invalid. Can you clarify? `-0x1n` seems fine to me AFAICT from the spec

Reply:
This is error, probably should be `legacyoctbigint`, but not sure if this should error early